app/helpers/application_helper.rb

```diff
@@ -199,26 +199,10 @@ def url_for_db(db, action = "show", item = nil) # Default action is show
   def view_to_url(view, parent = nil)
     association = view.scoped_association
     # Handle other sub-items of a VM or Host
-    case view.db
-    when "AdvancedSetting"  then association = "advanced_settings"
-    when "CloudNetwork"     then association = "cloud_networks"
-    when "OrchestrationStackOutput" then association = "outputs"
-    when "OrchestrationStackParameter" then association = "parameters"
-    when "OrchestrationStackResource"  then association = "resources"
-    when "Filesystem"       then association = "filesystems"
-    when "FirewallRule"     then association = "firewall_rules"
-    when "GuestApplication" then association = "guest_applications"
-    when "Patch"            then association = "patches"
-    when "RegistryItem"     then association = "registry_items"
-    when "ScanHistory"      then association = "scan_histories"
-    when "SystemService"
-      case parent.class.base_class.to_s.downcase
-      when "host" then association = "host_services"
-      when "vm"   then association = @lastaction
-      end
-    end
+    association = view_to_association(view.db, parent)
```

Comment:
oh my :open_mouth:  

Also I had hoped this patch would mean I won't have to touch this code [here](https://github.com/ManageIQ/manageiq/pull/6839/files#diff-127a850b7d7b20173ad3fba88f8a25e3R205) but I can't think of how to do that now.

Anyway I don't know if this patch is still worth the risk, it but I gave it another go.
Please review @chessbyte @jrafanie @abonas